src/js/actions/tools.js

```diff
@@ -417,7 +417,13 @@ define(function (require, exports) {
      * @return {Promise} Resolves to tool change
      */
     var selectTool = function (nextTool) {
-        var toolStore = this.flux.store("tool");
+        var toolStore = this.flux.store("tool"),
+            currentTool = toolStore.getCurrentTool();
+
+        // If same tool is being selected, make this a no-op
```

Comment:
I'm not sure that making the whole select function a no-op is what we want, but I want to test masks with this a bit before i press the button.